src/pocketmine/event/block/SignChangeEvent.php

```diff
@@ -79,7 +79,7 @@ public function getLine($index){
 	 * @param string[] $lines
 	 */
 	public function setLines(array $lines){
-		if(count($lines) > 4){
+		if(count($lines) === 4){
```

Comment:
and that will cause potentially unexpected behaviour by overwriting the remaining lines. That is why I said use setLine() instead.

You could have also used `array_pad()` instead of that awful looking micro optimized code.